core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ApacheSenderFactory.java

```diff
@@ -0,0 +1,22 @@
+package com.microsoft.applicationinsights.internal.channel.common;
+
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.reflect.ClassDataUtils;
+
+/**
+ * Created by gupele on 6/4/2015.
+ */
+final class ApacheSenderFactory {
+    public ApacheSender create() {
+        if (!ClassDataUtils.INSTANCE.verifyClassExists("org.apache.http.conn.HttpClientConnectionManager")) {
+            String errorMessage = String.format("Found an old version of HttpClient jar, for best performance consider upgrading to version 4.3+");
```

Comment:
Why using String.format if you're not formatting anything?
Actually you can just pass the message (`"Found an old..."`) directly to `InternalLogger.INSTANCE.warn()`